httpx/gzip_server.go

```diff
@@ -34,22 +33,13 @@ func (c *CompressionRequestReader) ServeHTTP(w http.ResponseWriter, r *http.Requ
 	for _, enc := range strings.Split(r.Header.Get("Content-Encoding"), ",") {
 		switch enc = strings.TrimSpace(enc); enc {
 		case "gzip":
-			var b bytes.Buffer
 			reader, err := gzip.NewReader(r.Body)
 			if err != nil {
 				c.ErrHandler(w, r, err)
 				return
 			}
-
-			if _, err := io.Copy(&b, reader); err != nil { //nolint:gosec // FIXME
```

Comment:
I guess the fixme comment is related to unlimited body reading, but I don't see why we should read everything here in the first place.